influxdb3_write/src/write_buffer/buffer_segment.rs

```diff
@@ -303,10 +329,22 @@ impl DatabaseBuffer {
         segment_key: &PartitionKey,
         table_batch: TableBatch,
     ) {
+        if !self.table_buffers.contains_key(&table_name) {
+            // TODO: this check shouldn't be necessary. If the table doesn't exist in the catalog
```

Comment:
I created #24955 to follow up and check what's going on here.

Reply:
Does this happen without your changes at all?

Reply:
Without my changes, the limit test passes, but that's because the buffer doesn't look for the table in the db_schema. So it looks to me like the limit check is hit, but it still calls to buffer the write, which passes on the old code path because it doesn't look into the catalog. That would be a bug because the buffered write would make it into the WAL and into the in-memory buffer without a corresponding catalog update. At least, that's what I think is going on here.